pkg/cmd/kafka/consumergroup/resetoffset/reset_offset.go

```diff
@@ -0,0 +1,246 @@
+		return cmdutil.FilterValidTopicNameArgs(f, toComplete)
+	})
+
+	flagutil.EnableOutputFlagCompletion(cmd)
+	flagutil.EnableStaticFlagCompletion(cmd, "offset", flagutil.ValidOffsets)
+
+	return cmd
+}
+
+// nolint:funlen
+func runCmd(opts *Options) error {
+
+	logger, err := opts.Logger()
+	if err != nil {
+		return err
+	}
+
+	conn, err := opts.Connection(connection.DefaultConfigRequireMasAuth)
+	if err != nil {
+		return err
+	}
+
+	api, kafkaInstance, err := conn.API().KafkaAdmin(opts.kafkaID)
+	if err != nil {
+		return err
+	}
+
+	ctx := context.Background()
+
+	offsetResetParams := kafkainstanceclient.ConsumerGroupResetOffsetParameters{
+		Offset: opts.offset,
+	}
+
+	if opts.value != "" {
+		offsetResetParams.Value = &opts.value
+	}
+
+	if opts.topic != "" {
+		topicToReset := kafkainstanceclient.TopicsToResetOffset{
+			Topic: opts.topic,
+		}
+
+		if len(opts.partitions) != 0 {
+			topicToReset.Partitions = &opts.partitions
+		}
+
+		topicsToResetArr := []kafkainstanceclient.TopicsToResetOffset{topicToReset}
+
+		offsetResetParams.Topics = &topicsToResetArr
+	}
+
+	a := api.GroupsApi.ResetConsumerGroupOffset(ctx, opts.id).ConsumerGroupResetOffsetParameters(offsetResetParams)
+
+	if !opts.skipConfirm {
+
+		var confirmReset bool
+		opts.localizer.MustLocalize("kafka.consumerGroup.resetOffset.input.confirmReset.message", localize.NewEntry("ID", opts.id))
+		promptConfirmReset := &survey.Confirm{
+			Message: opts.localizer.MustLocalize("kafka.consumerGroup.resetOffset.input.confirmReset.message", localize.NewEntry("ID", opts.id)),
+		}
+
+		if err = survey.AskOne(promptConfirmReset, &confirmReset); err != nil {
+			return err
+		}
+		if !confirmReset {
+			logger.Debug(opts.localizer.MustLocalize("kafka.consumerGroup.resetOffset.log.debug.cancelledReset"))
+			return nil
+		}
+	}
+
+	updatedConsumers, httpRes, err := a.Execute()
+
+	if err != nil {
+
+		if httpRes == nil {
+			return err
+		}
+
+		operationTmplPair := localize.NewEntry("Operation", "reset offset")
+
+		switch httpRes.StatusCode {
+		case http.StatusUnauthorized:
+			return errors.New(opts.localizer.MustLocalize("kafka.consumerGroup.common.error.unauthorized", operationTmplPair))
+		case http.StatusForbidden:
+			return errors.New(opts.localizer.MustLocalize("kafka.consumerGroup.common.error.forbidden", operationTmplPair))
+		case http.StatusInternalServerError:
+			return errors.New(opts.localizer.MustLocalize("kafka.consumerGroup.common.error.internalServerError"))
+		case http.StatusServiceUnavailable:
+			return errors.New(opts.localizer.MustLocalize("kafka.consumerGroup.common.error.unableToConnectToKafka", localize.NewEntry("Name", kafkaInstance.GetName())))
+		default:
+			return err
+		}
+	}
+
+	defer httpRes.Body.Close()
+
+	logger.Info(opts.localizer.MustLocalize("kafka.consumerGroup.resetOffset.log.info.successful", localize.NewEntry("ConsumerGroupID", opts.id), localize.NewEntry("InstanceName", kafkaInstance.GetName())))
```

Comment:
I wonder should be start breaking long logs onto multiple lines (the code, not the content).